### PR TITLE
fix: applyCustomKcats with customKcats structure input

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
- fix
  - `applyCustomKcats` should be able to directly take a customKcat structure as input (solves #303)

**I hereby confirm that I have:**

- [x] Selected `develop` as a target branch (top left drop-down menu)

### CODE REVIEW

**Comment by edkerk** [comment]
@ae-tafur could you run `runtests('geckoCoreFunctionTests/testKcats_tc0011')` to test this on your computer?

File: test/unit_tests/geckoCoreFunctionTests.m [hunk 1]
```diff
@@ -405,6 +405,18 @@ function testKcats_tc0011(testCase)
     ecModel=getKcatAcrossIsoenzymes(ecModel);
     verifyEqual(testCase,ecModel.ec.kcat, [1;1;10;10;1008;1009;1010;100;1011])
 
+    %Check applyCustomKcats
+    test = applyCustomKcats(ecModel);
```
Comment (ae-tafur) [inline test/unit_tests/geckoCoreFunctionTests.m:409]:
```suggestion
    test = applyCustomKcats(ecModel, [], adapter);
```

File: test/unit_tests/geckoCoreFunctionTests.m [hunk 2]
```diff
@@ -405,6 +405,18 @@ function testKcats_tc0011(testCase)
...
+    verifyEqual(testCase,test.ec.kcat, [100;200;50;50;100;200;1010;100;50])
+
+    customKcats.proteins = {'P3'; 'P1 + P2'; ''};
+    customKcats.kcat     = [200; 100; 50];
+    customKcats.rxns     = {'';'';'R2_REV, R5'};
+
+    test = applyCustomKcats(ecModel, customKcats);
```
Comment (ae-tafur) [inline test/unit_tests/geckoCoreFunctionTests.m:416]:
```suggestion
    test = applyCustomKcats(ecModel, customKcats, adapter);
```

**Comment by ae-tafur** [comment]
> @ae-tafur could you run `runtests('geckoCoreFunctionTests/testKcats_tc0011')` to test and review this on your computer?

With the changes suggested, works:

```
ans = 

  TestResult with properties:

          Name: 'geckoCoreFunctionTests/testKcats_tc0011'
        Passed: 1
        Failed: 0
    Incomplete: 0
      Duration: 0.7229
       Details: [1×1 struct]

Totals:
   1 Passed, 0 Failed, 0 Incomplete.
   0.72288 seconds testing time.
```

**Review by ae-tafur — APPROVED** [review]

**Pull request merged**